substratools/algo.py

```diff
@@ -104,312 +67,96 @@ def _parser_add_default_arguments(parser):
-
-        This method is called by the algorithm wrapper when the fake data mode
-        is enabled. In fake data mode, `X` and `y` input args have been
-        replaced by the opener fake data.
-
-        By default, it only calls directly `Algo.train()` method. Override this
-        method if you want to implement a different behavior.
-        """
-        return self.train(*args, **kwargs)
-
-    def _predict_fake_data(self, *args, **kwargs):
-        """Predict model fake data mode.
-
-        This method is called by the algorithm wrapper when the fake data mode
-        is enabled. In fake data mode, `X` input arg has been replaced by
-        the opener fake data.
-
-        By default, it only calls directly `Algo.predict()` method. Override
-        this method if you want to implement a different behavior.
-        """
-        return self.predict(*args, **kwargs)
-
-
-class AlgoWrapper(object):
-    """Algo wrapper to execute an algo instance on the platform."""
-
-    _INTERFACE_CLASS = Algo
-
-    def __init__(self, interface, workspace, opener_wrapper=None):
+    def __init__(
+        self, interface: GenericAlgo, workspace: GenericAlgoWorkspace, opener_wrapper: Optional[opener.OpenerWrapper]
+    ):
         assert isinstance(interface, self._INTERFACE_CLASS)
         self._workspace = workspace
-        self._opener_wrapper = opener_wrapper or opener.load_from_module(workspace=self._workspace)
+        self._opener_wrapper = opener_wrapper
         self._interface = interface
-
         self._interface.chainkeys_path = self._workspace.chainkeys_path
 
-    def _assert_output_exists(self, path, key):
-
-        if os.path.isdir(path):
-            raise exceptions.NotAFileError(f"Expected output file at {path}, found dir for output `{key}`")
-        if not os.path.isfile(path):
-            raise exceptions.MissingFileError(f"Output file {path} used to save argument `{key}` does not exists.")
-
-    def _load_models_paths(self):
-        if not self._workspace.input_model_paths:
-            return []
-        return self._workspace.input_model_paths
+    def _assert_outputs_exists(self, outputs: Dict[str, str]):
+        for key, path in outputs.items():
+            if os.path.isdir(path):
+                raise exceptions.NotAFileError(f"Expected output file at {path}, found dir for output `{key}`")
+            if not os.path.isfile(path):
+                raise exceptions.MissingFileError(f"Output file {path} used to save argument `{key}` does not exists.")
 
     @utils.Timer(logger)
-    def train(self, rank=0, fake_data=False, n_fake_samples=None):
-        """Train method wrapper."""
-        # load data from opener
-        X = self._opener_wrapper.get_X(fake_data, n_fake_samples)
-        y = self._opener_wrapper.get_y(fake_data, n_fake_samples)
+    def task_launcher(self, method_name: str, rank: int = 0, fake_data: bool = False, n_fake_samples: int = None):
+        """Launch Tasks"""
```

Comment:
```suggestion
    def execute(self, method_name: str, rank: int = 0, fake_data: bool = False, n_fake_samples: int = None):
        """Execute a compute task"""
```

Reply:
fixed in https://github.com/Substra/substra-tools/pull/49/commits/ce59796d4e51b35708087646c5d93287ef204848